Google.Api.Gax.Grpc/AsyncResponseStream.cs

```diff
@@ -0,0 +1,79 @@
+﻿/*
+ * Copyright 2019 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// An adapter from the gRPC stream representation (<see cref="IAsyncStreamReader{T}"/>) to <see cref="IAsyncEnumerable{T}"/>
+    /// and <see cref="IAsyncEnumerator{T}"/>.
+    /// </summary>
+    /// <remarks>
+    /// This type implements both of the standard asynchronous sequence interfaces for simplicity of use:
+    /// <list type="bullet">
+    ///   <item>C# 8 users can use  <c>await foreach</c> because it implements <see cref="IAsyncEnumerable{T}"/></item>
+    ///   <item>It's compatible with the System.Linq.Async package for query transformations.</item>
+    ///   <item>Pre-C# 8 users who will be calling <see cref="MoveNextAsync()" /> and <see cref="Current"/> directly don't need
+    ///   to call <see cref="GetAsyncEnumerator(CancellationToken)"/>.</item>
+    /// </list>
+    /// Note that <see cref="GetAsyncEnumerator(CancellationToken)"/> can only be called once per instance due to the "only iterate once"
```

Comment:
This warning might be better in the summary.

Reply:
I'm in two minds about that, but happy to go that way.